lib/middleware.js

```diff
@@ -0,0 +1,67 @@
+/* jshint node: true */
+'use strict';
+
+var Rule = require('./rule.js');
+
+module.exports = Middleware;
```

Comment:
"Middleware" in Node would normally be interpreted as being for [Express](http://expressjs.com/en/guide/using-middleware.html)...maybe use a more specific name for this file?